ansible/roles/network/tasks/main.yml

```diff
@@ -39,16 +43,86 @@
     dest: /boot/network.ini-sample
   tags:
     - touches_boot_partition
+  when: ansible_distribution_major_version|int != 9
 
 - name: Activate network manager
   command: systemctl enable screenly-net-manager.service chdir=/etc/systemd/system
+  when: ansible_distribution_major_version|int != 9
 
 - name: Activate network watchdog
   command: systemctl enable screenly-net-watchdog.timer chdir=/etc/systemd/system
+  when: ansible_distribution_major_version|int != 9
 
 - name: Install cronjob for watchdog
   cron:
     name: Screenly Net Watchdog
     minute: 0
     job: /usr/bin/python /usr/sbin/screenly_net_watchdog.py
     state: absent
+  when: ansible_distribution_major_version|int != 9
+
+
+# Use resin-wifi-connect if Stretch
+
+- name: Install NetworkManager
+  apt:
+    name: network-manager
+    state: present
+  when: ansible_distribution_major_version|int == 9
+
+- name: Disable network manager
+  command: systemctl disable screenly-net-manager.service
+  when: ansible_distribution_major_version|int == 9
+
+- name: Disable network watchdog
+  command: systemctl disable screenly-net-watchdog.timer
+  when: ansible_distribution_major_version|int == 9
+
+- name: Stop dhcpcd
+  command: systemctl stop dhcpcd
+  when: ansible_distribution_major_version|int == 9
+
+- name: Disable dhcpcd
+  command: systemctl disable dhcpcd
+  when: ansible_distribution_major_version|int == 9
+
+- name: Activate NetworkManager
+  command: systemctl enable NetworkManager
+  when: ansible_distribution_major_version|int == 9
+
+- name: Download resin-wifi-connect release
+  get_url:
+    url: https://github.com/resin-io/resin-wifi-connect/releases/download/v3.0.4/wifi-connect-v3.0.4-linux-rpi.tar.gz
+    dest: /home/pi/resin-wifi-connect.tar.gz
```

Comment:
It's better to use `/tmp` to avoid writes to the SD card.

Reply:
I agree and I did so initially, but it caused some problems. I mentioned this in one of the reports:
>  if I unpack `resin-wifi-connect` into `/tmp` folder, then the installation of apt packages does not work. I could not find an explanation for this, but now I'm just unpacking it in another directory.

Reply:
Interesting. I'm not sure how big this package is. Since `/tmp` is tmpfs, perhaps it exceeds the amount of available RAM. In any case, ignore this comment then.